Upgrading react-file-type-icons package for v9

bigbadcapers opened the pull request:
## Summary

Moves `@fluentui/react-file-type-icons` from the v8 layout (`packages/react-file-type-icons`) to the v9 layout (`packages/react-components/react-file-type-icons/library`), introduces a `FileTypeIcon` component built on the v9 hook/slot/Griffel pattern, and keeps the existing utility surface (`initializeFileTypeIcons`, `getFileTypeIconProps`, `getFileTypeIconAsHTMLString`, `getFileTypeIconAsUrl`, `FileIconType`, `FileTypeIconMap`) intact for current consumers. The package version jumps to `9.0.0` while keeping its npm name.

## Notes for v8 consumers upgrading to 9.x

- **Same npm name, major version bump.** The package is still published as `@fluentui/react-file-type-icons`, but starts at `9.0.0`. Apps pinned to `^8` will not auto-update — opt-in is required.
- **No more `v8` release path.** The Nx project is tagged `vNext`, so the package no longer participates in the v8 beachball release; the v8 scope assertion in `scripts/beachball/src/config.test.ts` was updated accordingly. There is no `master`-based maintenance lane for `8.x`.
- **SSR robustness in `getFileTypeIconSuffix`.** The old implementation evaluated `win ??= window`, which threw under SSR. The new implementation falls back to `devicePixelRatio = 1` when no `window` is available, so callers running on the server now get the `1x` suffix instead of an exception.
- **`initializeFileTypeIcons` options.** The `options?: Partial<IIconOptions>` argument is still accepted and the `disableWarnings` flag is honored. Re-registering an already-registered file type icon now logs a `console.warn` (suppressible via `{ disableWarnings: true }`). Note: unlike the v8 `@fluentui/style-utilities` `registerIcons`, the v9 implementation **overwrites** the existing record instead of skipping it. This is intentional and is covered by tests.
- **Global icon store shape preserved.** The new `initializeFileTypeIcons` writes to `globalThis.__globalSettings__.icons` using the same `{ code, subset: { mergeImageProps: true } }` record shape as v8. A v8 host still using `@fluentui/react`'s `<Icon iconName="docx16_svg" />` will continue to resolve file type icons correctly regardless of module load order.
- **`src/version.ts` is gone.** v9 packages do not call `setVersion(...)`, so the v8-style duplicate-package warning will not fire if `8.x` and `9.x` copies end up bundled side by side. Avoid installing both simultaneously.

## Checklist

* [x] Code is up-to-date with the `master` branch
* [x] Your changes are covered by tests (if possible)
* [x] Package based on existing v8 @fluentui/react-file-type-icons with additional error checks, better documentation and testing.
* [x] You've run `yarn change` locally

bigbadcapers committed:
Adding `portfolio` (Microsoft Project item type) to the list of non-fileextension based filetype icons.

bigbadcapers committed:
More filetype support.

bigbadcapers committed:
update CDN url and typo

bigbadcapers committed:
yarn change

bigbadcapers committed:
API update (from yarn nx run-many -t build)

bigbadcapers committed:
Merge branch 'master' into caperez/sep-filetype-icon-update

bigbadcapers committed:
Ran `yarn nx format`

bigbadcapers committed:
Merge branch 'master' into caperez/sep-filetype-icon-update

bigbadcapers committed:
Merge branch 'caperez/sep-filetype-icon-update' of https://github.com/bigbadcapers/fluentui into caperez/sep-filetype-icon-update

bigbadcapers committed:
Changing FileIconType-based icon searching to use the same scalable method used for file extensions

bigbadcapers committed:
Merge branch 'upstream-master' into caperez/newTypeMethod

# Conflicts (merged and resolved):
#	packages/react-file-type-icons/src/FileIconType.ts
#	packages/react-file-type-icons/src/FileTypeIconMap.ts
#	packages/react-file-type-icons/src/getFileTypeIconProps.ts

bigbadcapers committed:
removing stray outdated yarn change files

bigbadcapers committed:
yarn change

bigbadcapers committed:
Ran `yarn nx format`

bigbadcapers committed:
post-refactoring cleanup of an unused type

bigbadcapers committed:
fixing error reported by `lint`: error  The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype  guard-for-in

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'upstream-master' into caperez/newTypeMethod

# Conflicts:
#	packages/react-file-type-icons/src/FileIconType.ts
#	packages/react-file-type-icons/src/getFileTypeIconProps.ts

bigbadcapers committed:
adding album entry to filetypeicon mapping

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'microsoft:master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
first drop of creating a Fluent v9 version of the react-file-type-icons package. VSGHCP(CS4.5) following guidance from https://github.com/microsoft/fluentui/tree/master/docs/react-v9/contributing

bigbadcapers committed:
refreshing filetype icon references in other storybooks to new icon style (TO-DO: These should be updated to use getFileTypeIconProps with default CDN URL location to perma-fix)

bigbadcapers committed:
more touch-ups in other places that reference the CDN url

bigbadcapers committed:
verified new file-type-icons-preview package is rendering icons in the storybook as expected

bigbadcapers committed:
simplified documentation.

bigbadcapers committed:
simplifying getFileTypeIconAsURL and getFileTypeIconAsHTMLString

bigbadcapers committed:
updated storybook / tests to evaluate helper functions

bigbadcapers committed:
organizing storybook place in nav

bigbadcapers committed:
Merge branch 'master' into caperez/newTypeMethod

bigbadcapers committed:
Merge remote-tracking branch 'origin/caperez/newTypeMethod' into caperez/filetype-icons-for-v9

bigbadcapers committed:
update getFileTypeIconProps to use same mapping file FileTypeIconMap.ts to fetch by FileIconType instead of a separate enum

bigbadcapers committed:
new documentation pass 1 (broken)

bigbadcapers committed:
documentation rendering ok, ready to condense

bigbadcapers committed:
refining FileIconType and storybook

bigbadcapers committed:
story condensation snapshot

bigbadcapers committed:
condensation 2

bigbadcapers committed:
improving alt tag generation in the control itself, removing a11y story since it doesnt add much

bigbadcapers committed:
ran all pre-checks from plan-testingAndPromotionEnhanced. ready to start package rename

bigbadcapers committed:
package move to remove -preview suffix

bigbadcapers committed:
first test passed after package rename

bigbadcapers committed:
API update and undoing newFileTypeMethod in v8 package

bigbadcapers committed:
package semver cleanup, build pass, documentation simplification

bigbadcapers committed:
error checking and optimizations

bigbadcapers committed:
initializeFileTypeIcons with alternate CDN URL is now honored by new methods

bigbadcapers committed:
improved icon size checking, removing unnecessary public exports, fixing tests

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
cleanup stray API file

bigbadcapers committed:
Merge branch 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
changing v8 package name and adding publishConfig so it emits to the same npm

bigbadcapers committed:
updating swc packages and lint tweaks

bigbadcapers committed:
yarn change

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
remove private flag from package.json for react-file-type-icons

bigbadcapers committed:
Merge branch 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
deduplicating yarn lock and solving plugin issue in react-utilities .swcrc

bigbadcapers committed:
yarn change for react-utilities

bigbadcapers committed:
npx yarn-deduplicate --strategy fewer

bigbadcapers committed:
Add path mapping for react-file-type-icons-v8 in tsconfig.base.all.json

bigbadcapers committed:
re-running yarn nx format:write

bigbadcapers committed:
undoing removal of indent rule

bigbadcapers committed:
plugin swc-plugin-de-indent-template-literal is crashing, removing

bigbadcapers committed:
yarn change for experimental swc.rs indentation plugin removal

bigbadcapers committed:
fix missing param in documentation for getFileTypeIcon

bigbadcapers committed:
Add missing dependency and update styles in file type icon stories

bigbadcapers committed:
lint and yarn change again

bigbadcapers committed:
undoing overeager deletion of a file i had no business touching

bigbadcapers committed:
fixing compilation config issues - no change

bigbadcapers committed:
Refactor UrlAndHtml component export and update story parameters

bigbadcapers committed:
Add device pixel ratio state and effect to UrlAndHtml component to pass ci/cd test

bigbadcapers committed:
yarn nx format:write prettier

bigbadcapers committed:
dead end ci/cd solving ci/main no dice

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
first late undo

bigbadcapers committed:
Merge branch 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
attempted reset

bigbadcapers committed:
yarn change in shared-contexts

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
chore: update package dependencies for react and charting components

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
feat: add public docsite setup for v8 version of file-type-icons package

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers commented on packages/react-file-type-icons/package.json:
```diff
@@ -1,7 +1,7 @@
-  "name": "@fluentui/react-file-type-icons",
-  "version": "8.15.2",
-  "description": "Fluent UI React file type icon set.",
+  "name": "@fluentui/react-file-type-icons-v8",
```
Agreed, check out the change a few lines down on 16..19. The change in line 2 is scoped to the _yarn/npx workspace_ so everything compiles but the intent is that outwardly, the package in this folder location keeps publishing under the 8.x.x major semver, while the new package at `packages/react-components/react-file-type-icons` publishes unto the same @fluentui/react-file-type-icons NPM feed at 9.x.x 

Is there a better way to deliver a v9 compatible package, ie by making changes "in-place" in the package here such that it conforms to v9 requirements and semantics needed by the folks in the Teams convo and others in the future? If so, my preference would be to overlay that here instead of having two packages for which i would have to maintain the FileTypeIconMap.ts and FileIconType enums on, for sure. Very happy to adjust if there is a better pattern to apply here to upgrades/migrations.

*the usage of the publishConfig attr is what i found researching how to do this, but i have ofc not been able to test live. will [this approach](https://docs.npmjs.com/cli/v7/configuring-npm/package-json) work with our CI pipeline?

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers commented:
This is a v9 version of the [@fluentui/react-file-type-icons](https://github.com/microsoft/fluentui/tree/master/packages/react-file-type-icons) package which is v8, using the publishConfig directive so it ships to npm under the 9.x.x. semver at the same time the old package can keep shipping at the 8.x.x semver in the same namespace. 

Built to support request from @layershifter to bring aligned filetype icon representation to the latest Teams experience.

bigbadcapers committed:
missing comma after merge

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
fixing overwritten package name differentiation, retaining publishConfig to publish both v8.x.x and v9.x.x packages if needed

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
added correct config in syncpack semvergroups to support sim-shipping

bigbadcapers committed:
Merge branch 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
fixing syncpack config

bigbadcapers committed:
update package name for file type icons to v8 version

bigbadcapers committed:
fixing name error - undoing

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
fix: correct JSON formatting in package.json by adding missing comma

bigbadcapers committed:
chore: update yarn.lock

bigbadcapers committed:
chore: update API file for react-file-type-icons

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
feat: add targets configuration for generate-api in project.json

bigbadcapers committed:
Merge branches 'caperez/filetype-icons-for-v9' and 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
initial ship of v9 in v8

bigbadcapers committed:
feat: enhance FileTypeIcon documentation and add migration/playground stories

bigbadcapers committed:
feat: add Storybook support for react-file-type-icons and mock set-version module

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
feat(nx): add skills for generating code, managing plugins, running tasks, and exploring workspaces

- Introduced `nx-generate` skill for scaffolding projects and code generation using Nx generators.
- Added `nx-plugins` skill to discover and install Nx plugins for various frameworks.
- Created `nx-run-tasks` skill to assist in executing tasks within an Nx workspace.
- Developed `nx-workspace` skill for exploring workspace structure, project configurations, and available targets.
- Added references for affected projects in the workspace.
- Updated VSCode settings for improved terminal history and session management.
- Enhanced guidelines for working with Nx, emphasizing the use of skills for scaffolding and exploration.
- Updated Storybook configuration to ensure visibility of specific component documentation.
- Added mock for `@fluentui/set-version` to prevent errors in Storybook.

bigbadcapers committed:
feat: optimize file type icon mapping by introducing FileIconTypeToIconNameMap

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into user/caperez/v9-fit-in-8

bigbadcapers committed:
in progress of creating v9-in-v8 version of file type icons package

bigbadcapers committed:
chore: update TypeScript configuration to include ignoreDeprecations option

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into user/caperez/v9-fit-in-8

bigbadcapers committed:
v1 of storybook integration, but at the wrong spot

bigbadcapers committed:
feat: update Storybook integration for v9 and remove deprecated FileTypeIcon stories

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into user/caperez/v9-fit-in-8

bigbadcapers committed:
Merge branch 'user/caperez/v9-fit-in-8' into caperez/filetype-icons-for-v9 (match source tree)

bigbadcapers committed:
yarn change

bigbadcapers committed:
fix(Fluent v9 Storybook and FileTypeIcon.Basic.Example): update filetype icon descriptions and types

bigbadcapers committed:
fix: update package dependencies and clean up tsconfig files

bigbadcapers committed:
storybook build artiface cleanup

bigbadcapers committed:
testing and CI fixups

bigbadcapers committed:
fix: improve window handling and type annotations in file type icon functions

bigbadcapers committed:
fix: add path for react-file-type-icons stories in tsconfig

bigbadcapers committed:
fix: correct indentation in JSDoc parameter comment in webpackFinal function

bigbadcapers committed:
fix: enhance window resolution handling in getFileTypeIconSuffix function

bigbadcapers committed:
adding line required to properly do liniting in the CI. should this mean the package should be moved into the packages/react-components folder? and will that have negative effects for v8 usages?

bigbadcapers committed:
trying to solve CI failure: nx run react-file-type-icons-stories:lint

bigbadcapers committed:
fix: add path for react-file-type-icons to package.json workspaces

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

# Conflicts:
#	apps/public-docsite-v9/.storybook/preview.js
#	apps/public-docsite-v9/src/Icons/IconsBundleIcon.stories.tsx
#	packages/react-file-type-icons/package.json
#	packages/react-file-type-icons/src/getFileTypeIconProps.ts

bigbadcapers committed:
feat: add file type icon stories for default, custom base URL, size and formats, and V8 utility interop

bigbadcapers committed:
feat(react-file-type-icons): move package under react-components

bigbadcapers committed:
chore(react-file-type-icons): finish validation fixes

bigbadcapers committed:
chore(react-file-type-icons): align story file names

bigbadcapers committed:
fix(storybook): support Windows sandbox source generation

bigbadcapers committed:
fix(react-file-type-icons): avoid JSX runtime in legacy utility builds

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
refactor: clean up VSCode settings and remove unused tasks; update Storybook webpack tests for consistency

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
refactor: simplify import initialization in modifyImportsPlugin

bigbadcapers committed:
fix: register file type icons without style-utilities

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
feat(react-file-type-icons): refactor stories to use new file structure and restore deleted stories

bigbadcapers committed:
STORY feat(react-file-type-icons): enhance stories with descriptions and examples, add fallback behavior and playground

bigbadcapers committed:
STORY 2 feat(react-file-type-icons): update story descriptions and order for better clarity and navigation

bigbadcapers committed:
STORY 3 feat(react-file-type-icons): update story descriptions and names for clarity and consistency

bigbadcapers committed:
STORY 4

bigbadcapers committed:
STORY tidying 5

bigbadcapers committed:
story tidied

bigbadcapers committed:
params tidied

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
story and PR tweaks

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
API update

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
This branch showed lint problems when merged an unrelated commit, submitting with linter fix to ensure the GH Actions pass.

bigbadcapers committed:
chore: add linter fix configuration for @fluentui/react-headless-components-preview

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge branch 'caperez/filetype-icons-for-v9' of https://github.com/bigbadcapers/fluentui into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

tudorpopams commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/useFileTypeIcon_unstable.ts:
```diff
@@ -0,0 +1,60 @@
+  };
+
+  if (state.root.width === undefined) {
+    state.root.width = size;
```
**Blocker (CI):** `react-compiler-analyzer` is failing on this PR. The styles hook opts out with `'use no memo'`, but this hook does not and conditionally mutates `state.root.width`/`state.root.height` after slot creation. Please get the analyzer green or confirm the failure is unrelated before merge.

bigbadcapers replied:
Fixed at the root: the `width`/`height` defaults now flow through `getIntrinsicElementProps('img', { ref, width: size, height: size, ...rootProps, src, 'aria-label': ..., 'data-icon-name': ... })`, so there's no post-slot mutation. User-supplied `width`/`height` in `rootProps` still win because they spread after the defaults. As a side effect, the analyzer also flagged the `'use no memo'` in `useFileTypeIconStyles.styles.ts` as redundant — auto-removed. `react-compiler-analyzer` is green in CI.

tudorpopams commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/useFileTypeIcon_unstable.ts:
```diff
@@ -0,0 +1,60 @@
+
+  const options = { extension, type, size, imageFileType };
+  const iconProps = getFileTypeIconProps(options);
+  const src = getFileTypeIconAsUrl(options, baseUrl);
```
**DRY:** `getFileTypeIconProps(options)` (line 25) and `getFileTypeIconAsUrl(options, baseUrl)` both resolve the icon name + suffix — the latter re-runs `getFileTypeIconNameFromExtensionOrType` + `getFileTypeIconSuffix` inside `resolveFileTypeIconUrl`. That's the same work twice per render. Consider resolving once (e.g. derive `iconName`/`aria-label` and `src` from a single `resolveFileTypeIconUrl` call).

bigbadcapers replied:
Replaced both calls with a single `resolveFileTypeIconUrl({ extension, type, size, imageFileType }, baseUrl)` that returns `{ src, iconName, ariaLabel, usesPixelRatioDirectory }`. The component now resolves icon name + URL + aria-label in one pass. `getFileTypeIconProps` and `getFileTypeIconAsUrl` are kept as thin public-API wrappers for legacy callers (no public surface change — `etc/react-file-type-icons.api.md` is unchanged).

tudorpopams commented on packages/react-components/react-file-type-icons/library/src/initializeFileTypeIcons.tsx:
```diff
@@ -0,0 +1,140 @@
+    // SVGs scale well, so you can generally use the default image.
+    // 1.5x is a special case where both SVGs and PNGs need a different image.
+
+    fileTypeIcons[type + size + '_1.5x' + PNG_SUFFIX] = createFileTypeIconImage(
```
**DRY:** these six near-identical `createFileTypeIconImage(...)` blocks (default + `_1.5x` png/svg + `_2x`/`_3x`/`_4x` png) are easy to get subtly wrong — note `1.5x` exists for both png+svg while `2x/3x/4x` are png-only, which is hard to verify by eye. Prefer a data-driven loop over a `[{ dir, ext }]` table so the matrix is declared once.

bigbadcapers replied:
Replaced with a declarative table:

```ts
const ICON_VARIANTS = [
  { dir: '',      exts: ['png', 'svg'] }, // 1x
  { dir: '_1.5x', exts: ['png', 'svg'] },
  { dir: '_2x',   exts: ['png'] },
  { dir: '_3x',   exts: ['png'] },
  { dir: '_4x',   exts: ['png'] },
] as const;
```

`_initializeIcons` now iterates `iconTypes × ICON_VARIANTS × exts`. The asymmetric matrix is declared once and obvious by inspection. `initializeFileTypeIcons.test.tsx` (which asserts all the `docx16_*` / `docx16_1.5x_*` / `docx16_2x_png` / etc. keys) still passes unchanged.

tudorpopams commented on packages/react-components/react-file-type-icons/library/src/initializeFileTypeIcons.tsx:
```diff
@@ -0,0 +1,140 @@
+
+type IconRecords = Record<string, unknown> & {
+  __options: IIconOptions;
+  __remapped: Record<string, string>;
```
**KISS / YAGNI:** `__remapped` (here) and the `__callbacks__` state (`CALLBACK_STATE_PROP_NAME`, lines 23/27/131) are written but never read anywhere in this package. Only the `{ code, subset }` records under `__globalSettings__.icons` matter for the documented v8 `<Icon>` interop. Dropping the unused scaffolding shrinks the surface area of this re-implemented store.

bigbadcapers replied:
Dropped:

- `CALLBACK_STATE_PROP_NAME` const + the `[CALLBACK_STATE_PROP_NAME]: {}` writes in `moduleGlobalSettings` and `getGlobalSettings`.
- `__remapped: {}` initializer in `getIconSettings`.
- The `__remapped` field on the `IconRecords` type.

Kept `__options` (consumed by `disableWarnings`) and the `{ code, subset }` per-icon records (the documented v8 `<Icon>` interop). Tests still pass.

tudorpopams commented on packages/react-components/react-file-type-icons/library/src/fileTypeIconUrl.ts:
```diff
@@ -0,0 +1,38 @@
+  const { extension, size = DEFAULT_ICON_SIZE, type, imageFileType } = options;
+  const baseIconName = getFileTypeIconNameFromExtensionOrType(extension, type);
+  const baseSuffix = getFileTypeIconSuffix(size, imageFileType);
+  const suffixArray = baseSuffix.split('_');
```
**KISS (fragile coupling):** this reverse-engineers the dpr directory by `split('_')`-ing the output of `getFileTypeIconSuffix` and branching on `length === 2 | 3`. URL construction is now implicitly coupled to the exact suffix string format of another function; a format tweak there silently breaks URLs here. Returning the dpr directory directly (instead of re-parsing a string) would be clearer and more robust.

bigbadcapers replied:
Refactored. New `getDevicePixelRatioVariant(size, imageFileType)` in `getFileTypeIconProps.ts` returns `{ dprDir, ext }` directly. Both `getFileTypeIconSuffix` (which composes the suffix string from those parts) and `resolveFileTypeIconUrl` (which composes the URL) now share that helper. No string parsing, no implicit coupling — change one and the other follows automatically.

tudorpopams commented on packages/react-components/react-file-type-icons/library/src/getFileTypeIconProps.ts:
```diff
@@ -105,89 +81,36 @@ export function getFileTypeIconNameFromExtensionOrType(
-  win ??= window;
-  const devicePixelRatio: number = win.devicePixelRatio;
+  const resolvedWindow =
+    win ?? (typeof globalThis !== 'undefined' && 'window' in globalThis ? globalThis.window : undefined);
```
**KISS / consistency:** this hand-rolls window detection (`typeof globalThis !== 'undefined' && 'window' in globalThis ? ...`). The package already depends on `@fluentui/react-utilities` and uses `canUseDOM()` in `initializeFileTypeIcons.tsx`. Reusing `canUseDOM()` here would be more readable and consistent.

bigbadcapers replied:
Done — `getFileTypeIconSuffix` / `getDevicePixelRatioVariant` now use `canUseDOM()` from `@fluentui/react-utilities`. Also dropped the unused `win?: Window` test-seam parameter (no internal callers, never appeared in `etc/*.api.md`). One small follow-up in `24f94270d3`: the lint rule wanted `globalThis.window` instead of bare `window` for the post-`canUseDOM()` access.

tudorpopams commented on change/@fluentui-react-file-type-icons-8ed04cc5-3dd7-452c-a5a2-c18c9ca7638a.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
**Info:** change `type` is `minor`, while the comment describes shipping `9.0.0` as a "major bump." The change-files check passes, so beachball accepts it — just confirm this is the intended semver outcome for the relocated package.

bigbadcapers replied:
Tried bumping to `"type": "major"` first, but `npx beachball check` rejected it: `react-file-type-icons/package.json` declares `disallowedChangeTypes: ["major", "prerelease"]` (carried over from the v8 package config), so `major` is invalid here. Reverted to `"type": "minor"` in `52f54bf4f5`. The published version still goes to `9.0.0` because `package.json` pins `"version": "9.0.0"` directly — the change-file `type` only governs the relative bump beachball would apply, which is moot for the first release at the new major. Happy to drop `"major"` from `disallowedChangeTypes` in a follow-up if we want it selectable for future releases.

tudorpopams reviewed (changes requested):
### Confidence Score: 60/100

Solid v8→v9 migration with good test/docs coverage, but CI is red and several DRY/KISS smells in the icon-registration and URL-resolution code warrant cleanup before merge. Inline comments below mark each finding.

#### Summary of findings

**Blocker**
- CI `react-compiler-analyzer` is failing — must be green (or confirmed unrelated) before merge.

**Warnings (DRY / KISS)**
- `_initializeIcons` dpr-variant repetition (DRY).
- Reimplemented v8 global icon store with dead `__remapped` / `__callbacks__` fields (KISS/YAGNI).
- `resolveFileTypeIconUrl` reverse-engineers the suffix via `split('_')` length 2/3 (KISS, fragile coupling).
- `useFileTypeIcon_unstable` resolves icon name/suffix twice (DRY).
- `getFileTypeIconSuffix` hand-rolls an SSR window guard instead of reusing `canUseDOM()` (KISS/consistency).

**Info**
- Change-file `type: minor` vs a comment describing a 9.0.0 major bump — confirm intended.

### Category Breakdown

| Category | Status | Notes |
|----------|--------|-------|
| Change file | PASS | 3 change files, well-described |
| V9 patterns | PASS | ForwardRefComponent, slots, assertSlots, mergeClasses w/ user className last |
| Dep layers | PASS | Only Tier 1/2 deps |
| SSR safety | PASS | canUseDOM() + window guards |
| Testing | PASS | Tests for utilities + component |
| API surface | PASS | etc/*.api.md added; shared-contexts api.md updated |
| Accessibility | PASS | aria-label plumbed; alt="" on decorative imgs |
| Security/Quality | WARN | DRY/KISS smells noted inline |
| Docs coverage | PASS | Stories, README, docsite + examples updated |

### Recommendation

**REQUEST_CHANGES** — primarily gated on the red `react-compiler-analyzer` check. The DRY/KISS items are non-blocking but worth tightening while this is the new long-lived v9 surface. Core architecture, layering, and test/doc coverage are good.

---
*Posted via the `/review-pr` skill.*

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
fix(modifyImports): ensure importMappings is defined with a fallback

bigbadcapers committed:
Address PR review feedback for react-file-type-icons v9

- Fix react-compiler-analyzer: move width/height defaults into slot.always() call (no post-slot mutation); remove redundant 'use no memo' from styles hook.

- DRY: useFileTypeIcon_unstable now uses single resolveFileTypeIconUrl() returning { src, iconName, ariaLabel }.

- DRY: replace six createFileTypeIconImage blocks in _initializeIcons with a declarative ICON_VARIANTS table.

- KISS/YAGNI: drop unused __remapped, __callbacks__, CALLBACK_STATE_PROP_NAME scaffolding from initializeFileTypeIcons.

- KISS: new shared getDevicePixelRatioVariant() helper drives both getFileTypeIconSuffix and resolveFileTypeIconUrl; no more split('_') reverse-parsing.

- KISS: use canUseDOM() in getFileTypeIconSuffix for SSR guard, consistent with initializeFileTypeIcons.

- Change file: minor -> major to match documented 9.0.0 jump.

bigbadcapers committed:
chore: add change file for babel-preset-storybook-full-source

bigbadcapers committed:
chore: revert react-file-type-icons change-type to minor (major is disallowed)

bigbadcapers committed:
Merge remote-tracking branch 'upstream/master' into caperez/filetype-icons-for-v9

bigbadcapers committed:
fix: use globalThis for window reference in getDevicePixelRatioVariant

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

tudorpopams reviewed (approved):
### Confidence Score: 98/100

Re-review of `441ae1f`: every actionable finding from my previous review has been addressed and CI is now fully green. Recommending approval.

#### Resolved since last review
- **[Blocker] `react-compiler-analyzer` CI** — now passing. The hook no longer mutates `state.root` post-slot (width/height are set inside `getIntrinsicElementProps`), and the `'use no memo'` opt-out was dropped. ✅
- **[DRY] Double resolution in `useFileTypeIcon_unstable`** — now a single `resolveFileTypeIconUrl(...)` call returning `{ src, iconName, ariaLabel }`. ✅
- **[DRY] `_initializeIcons` dpr repetition** — replaced with a declarative `ICON_VARIANTS` table iterated once. The png/svg-vs-dpr matrix is now stated in one place. ✅
- **[KISS] Dead global-store fields** — `__remapped` and `__callbacks__` removed; `moduleGlobalSettings` is now `{}`. ✅
- **[KISS] Fragile `split('_')` suffix coupling** — replaced by `getDevicePixelRatioVariant()` returning structured `{ dprDir, ext }`, shared by both `getFileTypeIconSuffix` and `resolveFileTypeIconUrl` so they stay in lock-step. ✅
- **[KISS] Hand-rolled SSR guard** — `getDevicePixelRatioVariant` now uses `canUseDOM()`. ✅

#### Info
- Change-file `type` remains `minor` (unchanged) — confirmed acceptable since `change-files` CI passes; just noting the 9.0.0 framing again for the record.
- Nice side effect: `getFileTypeIconAsUrl` / `getFileTypeIconAsHTMLString` now return a non-optional `string` (previous unreachable `undefined` path is gone) and both consumers were updated accordingly.

### Recommendation

**APPROVE** — clean v9 migration, good DRY/KISS follow-through, full test/doc coverage, and all checks green.

---
*Posted via the `/review-pr` skill.*

bigbadcapers committed:
Merge branch 'master' into caperez/filetype-icons-for-v9

layershifter commented on packages/react-components/react-shared-contexts/library/src/CustomStyleHooksContext/CustomStyleHooksContext.ts:
```diff
@@ -71,6 +71,7 @@ export type CustomStyleHooksContextValue = Partial<{
   useDropdownStyles_unstable: CustomStyleHook;
   useEmptySwatchStyles_unstable: CustomStyleHook;
   useFieldStyles_unstable: CustomStyleHook;
+  useFileTypeIconStyles_unstable: CustomStyleHook;
```
IMO it's redundant. Regular icons are also not there

layershifter commented on packages/react-components/react-file-type-icons/stories/src/FileTypeIcon/FileTypeIconDefault.stories.tsx:
```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { FileIconType, FileTypeIcon } from '@fluentui/react-file-type-icons';
+
+const columnStyle: React.CSSProperties = {
+  display: 'grid',
+  gridTemplateColumns: 'auto 1fr',
+  columnGap: 12,
+  rowGap: 8,
+  alignItems: 'center',
+};
```
Plz use `makeStyles()` there and in other places in stories

layershifter commented on packages/react-components/react-file-type-icons/library/README.md:
```diff
@@ -0,0 +1,51 @@
+
+## Getting started
+
+If you are using the v8 `Icon` component or composing with icon-name utilities, you can make all file type icons available by calling the `initializeFileTypeIcons` function from the `@fluentui/react-file-type-icons` package:
```
v8 mention is misleading there

layershifter commented on packages/react-components/react-file-type-icons/library/README.md:
````diff
@@ -0,0 +1,51 @@
+
+## Usage in code
+
+If you are using Fluent UI React, you can use the `Icon` component and pass in the corresponding icon properties to render a given icon.
+
+```tsx
+import { Icon } from '@fluentui/react/lib/Icon';
+import { getFileTypeIconProps } from '@fluentui/react-file-type-icons';
````
Snippet is invalid?

layershifter commented on packages/react-components/react-file-type-icons/stories/.storybook/main.js:
```diff
@@ -0,0 +1,23 @@
+    localConfig.resolve = localConfig.resolve || {};
+    localConfig.resolve.alias = {
+      ...(localConfig.resolve.alias || {}),
+      '@fluentui/react-file-type-icons$': path.resolve(__dirname, '../../library/src/index.ts'),
```
Is it really needed?

layershifter commented on packages/react-components/react-file-type-icons/library/src/initializeFileTypeIcons.tsx:
```diff
@@ -0,0 +1,119 @@
+
+export { DEFAULT_BASE_URL, ICON_SIZES } from './fileTypeIconUrl';
+
+export interface IIconOptions {
```
v9 package don't use `I` prefix for interfaces

layershifter commented on packages/react-components/react-file-type-icons/library/src/initializeFileTypeIcons.tsx:
```diff
@@ -0,0 +1,119 @@
+import { FileTypeIconMap } from './FileTypeIconMap';
+import { DEFAULT_BASE_URL, ICON_SIZES } from './fileTypeIconUrl';
+
+export { DEFAULT_BASE_URL, ICON_SIZES } from './fileTypeIconUrl';
```
Please remove this re-export in this file. If it's needed in `index.ts`, re-export directly from `./fileTypeIconUrl`

layershifter commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/FileTypeIcon.types.ts:
```diff
@@ -0,0 +1,26 @@
+    type?: IFileTypeIconOptions['type'];
+  };
+
+export type { ImageFileType };
```
Plz avoid type re-exports in non-barrel files

layershifter commented on packages/react-components/react-file-type-icons/library/src/index.ts:
```diff
@@ -1,4 +1,13 @@
 export { initializeFileTypeIcons } from './initializeFileTypeIcons';
+export type { IIconOptions } from './initializeFileTypeIcons';
```
```suggestion
export type { IconOptions } from './initializeFileTypeIcons';
```

layershifter commented on packages/react-components/react-file-type-icons/library/src/getFileTypeIconAsHTMLString.ts:
```diff
@@ -0,0 +1,21 @@
+import { DEFAULT_BASE_URL, resolveFileTypeIconUrl } from './fileTypeIconUrl';
+import type { IFileTypeIconOptions } from './getFileTypeIconProps';
```
```suggestion
import type { FileTypeIconOptions } from './getFileTypeIconProps';
```

layershifter commented on packages/react-components/react-file-type-icons/library/src/getFileTypeIconAsHTMLString.ts:
```diff
@@ -0,0 +1,21 @@
+ * @param options - Options used to resolve the file type icon HTML string.
+ * @param baseUrl - optionally provide a custom CDN base url to fetch icons from
+ */
+export function getFileTypeIconAsHTMLString(
```
Do we really need this? I would suggest to reduce API surface as much as possible

layershifter commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/useFileTypeIcon_unstable.ts:
```diff
@@ -0,0 +1,54 @@
+  const {
+    src,
+    iconName,
+    ariaLabel: resolvedAriaLabel,
```
TBH I don't think that extension is a good `aria-label`

layershifter commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/useFileTypeIcon_unstable.ts:
```diff
@@ -0,0 +1,54 @@
+        height: size,
+        ...rootProps,
+        src,
+        'aria-label': ariaLabel ?? resolvedAriaLabel,
```
Let's avoid `@fluentui/react-icons` strategy:

https://github.com/microsoft/fluentui-system-icons/blob/c154201b6194639bb2a2fb09292190b32da1bae3/packages/react-icons/src/utils/useIconState.tsx#L46-L48

layershifter commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/renderFileTypeIcon_unstable.tsx:
```diff
@@ -0,0 +1,15 @@
+  // eslint-disable-next-line @typescript-eslint/no-deprecated
+  const { slots, slotProps } = getSlots<FileTypeIconSlots>(state);
+
+  return React.createElement(slots.root, slotProps.root);
```
Plz don't use `React.createElement`, use JSX

layershifter commented on packages/react-components/react-file-type-icons/library/src/components/FileTypeIcon/renderFileTypeIcon_unstable.tsx:
```diff
@@ -0,0 +1,15 @@
+export const renderFileTypeIcon_unstable = (state: FileTypeIconState): JSXElement => {
+  assertSlots<FileTypeIconSlots>(state);
+  // eslint-disable-next-line @typescript-eslint/no-deprecated
+  const { slots, slotProps } = getSlots<FileTypeIconSlots>(state);
```
This should not be needed anymore

Hotell reviewed (commented):
this should be closed as we have new implementation in icons repo https://github.com/microsoft/fluentui-system-icons/pull/1095
